openerp/addons/base/ir/ir_qweb.py

```diff
@@ -59,7 +59,7 @@ def raise_qweb_exception(etype=None, **kw):
             e.qweb[k] = v
         # Will use `raise foo from bar` in python 3 and rename cause to __cause__
         e.qweb['cause'] = original
-        raise
+        raise etype(e.qweb.items)
 
```

Comment:
A much more elegant solution indeed! For some reason even though the message is more verbose the qweb data gets lost along the way somewhere...